test/test_proxy.py

```diff
@@ -0,0 +1,298 @@
+
+        return ProxyTestConfiguration.HTTP_PROXY_PORT
+
+    @staticmethod
+    def get_proxy_connection_type_for_test(test_type):
+        if test_type == ProxyTestType.FORWARDING:
+            return HttpProxyConnectionType.Forwarding
+        elif test_type == ProxyTestType.TUNNELING_DOUBLE_TLS or \
+                test_type == ProxyTestType.TUNNELING_HTTP or \
+                test_type == ProxyTestType.TUNNELING_HTTPS:
+            return HttpProxyConnectionType.Tunneling
+        else:
+            return HttpProxyConnectionType.Legacy
+
+    @staticmethod
+    def get_proxy_tls_connection_options_for_test(test_type):
+        if test_type == ProxyTestType.TUNNELING_DOUBLE_TLS:
+            tls_ctx_opt = TlsContextOptions()
+            tls_ctx_opt.verify_peer = False
+            tls_ctx = ClientTlsContext(tls_ctx_opt)
+            tls_conn_opt = tls_ctx.new_connection_options()
+            tls_conn_opt.set_server_name("localhost")
+            return tls_conn_opt
+        else:
+            return None
+
+    @staticmethod
+    def create_http_proxy_options_from_environment(test_type, auth_type):
+        return HttpProxyOptions(
+            ProxyTestConfiguration.get_proxy_host_for_test(
+                test_type,
+                auth_type),
+            ProxyTestConfiguration.get_proxy_port_for_test(
+                test_type,
+                auth_type),
+            tls_connection_options=ProxyTestConfiguration.get_proxy_tls_connection_options_for_test(test_type),
+            auth_type=auth_type,
+            auth_username=ProxyTestConfiguration.HTTP_PROXY_BASIC_AUTH_USERNAME,
+            auth_password=ProxyTestConfiguration.HTTP_PROXY_BASIC_AUTH_PASSWORD,
+            connection_type=ProxyTestConfiguration.get_proxy_connection_type_for_test(test_type))
+
+    @staticmethod
+    def get_tls_connection_options_for_test(test_type, host_name):
+        if test_type == ProxyTestType.FORWARDING or test_type == ProxyTestType.LEGACY_HTTP or test_type == ProxyTestType.TUNNELING_HTTP:
+            return None
+        else:
+            tls_ctx_opt = TlsContextOptions()
+            tls_ctx = ClientTlsContext(tls_ctx_opt)
+            tls_conn_opt = tls_ctx.new_connection_options()
+            tls_conn_opt.set_server_name(host_name)
+            return tls_conn_opt
+
+    @staticmethod
+    def get_uri_from_test_type(test_type):
+        if test_type == ProxyTestType.FORWARDING or test_type == ProxyTestType.LEGACY_HTTP or test_type == ProxyTestType.TUNNELING_HTTP:
+            return "www.example.com"
+        else:
+            return "www.amazon.com"
+
+    @staticmethod
+    def get_port_from_test_type(test_type):
+        if test_type == ProxyTestType.FORWARDING or test_type == ProxyTestType.LEGACY_HTTP or test_type == ProxyTestType.TUNNELING_HTTP:
+            return 80
+        else:
+            return 443
+
+
+class ProxyHttpTest(NativeResourceTest):
+
+    timeout = 10  # seconds
```

Comment:
just use our default `TIMEOUT` you already imported at the top of the file:
`from test import NativeResourceTest, TIMEOUT`